### DISCUSSION TITLE
[DONE] Building on Linux (WSL2)

### DISCUSSION THREAD

**Discussion by Jonny0815** [yamadapc/augmented-audio#598] [General]
Hello,

im trying to build this repo on wsl2, ive managed to build all dependancys by now (took some time) but now when actually trying to build the various apps its not working because they all use cocoa and this is macos only, as far as my understanding is

has anyone compiled this project on linux? am i lacking some basic knowlege? this is my first time trying to compile a bigger rust project (ive just done some basics without code from git)

any help is welcome
Johannes

**Comment by yamadapc** [comment]
Hey mate 👋, a lot of the apps indeed don't build on Linux, you can see the CI on what can be built / how to bypass the uncompatible crates.

In particular:
* https://github.com/yamadapc/augmented-audio/blob/master/scripts/install-ubuntu-dependencies.sh
* https://github.com/yamadapc/augmented-audio/blob/master/scripts/build-and-test-linux.sh

Have fun with what you want to play around with :) note this is very experimental and hobby project.

**Comment by Jonny0815** [comment]
thanks for you help, 

ive tried this but the plugin test host gui doesnt work, thats the one i was interrested in

for gui youre using iced, can you recommand crates for audio/midi io and vst hosting? any good learning resources you can point me to?
i was planing on playing around with your code to figgure stuff out but that doesnt seem to work on my end

and no worries, this is for education and hobby as well ✌️ 

thanks again
Johannes

**Comment by yamadapc** [comment]
What happens with the GUI?

The `vst` crate can be used for hosting. See `plugin-host-cli` and `plugin-host-lib` for the headless host.

**Comment by yamadapc** [comment]
I think on WSL there might be issues with `iced`; I was having issues running it on VirtualBox.
